word_language_model/generate.py

```diff
@@ -46,22 +46,48 @@
 with open(args.checkpoint, 'rb') as f:
     model = torch.load(f).to(device)
 model.eval()
+model.set_src_mask(None)
+model.set_tgt_mask(None)
+model.set_memory_mask(None)
 
 corpus = data.Corpus(args.data)
 ntokens = len(corpus.dictionary)
-hidden = model.init_hidden(1)
-input = torch.randint(ntokens, (1, 1), dtype=torch.long).to(device)
-
-with open(args.outf, 'w') as outf:
-    with torch.no_grad():  # no tracking history
-        for i in range(args.words):
-            output, hidden = model(input, hidden)
-            word_weights = output.squeeze().div(args.temperature).exp().cpu()
-            word_idx = torch.multinomial(word_weights, 1)[0]
-            input.fill_(word_idx)
-            word = corpus.dictionary.idx2word[word_idx]
-
-            outf.write(word + ('\n' if i % 20 == 19 else ' '))
-
-            if i % args.log_interval == 0:
-                print('| Generated {}/{} words'.format(i, args.words))
+
+if type(model).__name__ == "TransformerSeq2Seq":
+    input = torch.randint(ntokens, (1, 1), dtype=torch.long).to(device)
+    last_word = input[-1, :]
+
+    with open(args.outf, 'w') as outf:
+        with torch.no_grad():  # no tracking history
+            for i in range(args.words):
+                output = model(input, last_word)
+                word_weights = output.squeeze().div(args.temperature).exp().cpu()
+                word_idx = torch.multinomial(word_weights, 1)[0]
+                word = corpus.dictionary.idx2word[word_idx]
+                word_tensor = torch.ones(1).long().view(1,1).to(device)
```

Comment:
It seems like this is very similar to filling input with wordidx. Why is it necessary to do this multiplication etc. instead of filling input?